### ISSUE TITLE
[TypeError: Cannot read property '_location' of null] when window.close()

### ISSUE THREAD

**Issue by lijumathews** [tmpvar/jsdom#1542]
I'm trying to use window.close() within jsdom.jsdom and I get the the error
[TypeError: Cannot read property '_location' of null]
/node_modules/jsdom/lib/jsdom/browser/Window.js:134
return idlUtils.wrapperForImpl(idlUtils.implForWrapper(window._document)._location);
TypeError: Cannot read property '_location' of null
at Window.location (/node_modules/jsdom/lib/jsdom/browser/Window.js:134:79)
at callback (/node_modules/jsdom/lib/jsdom/browser/Window.js:466:40)
at Timer.listOnTimeout (timers.js:92:15)

var indexFile = fs.readFileSync( docRoot + "index.html", "utf-8");
var jsdom = require("jsdom");
    var options = {
        //html: indexFile,
        url: "",
        strictSSL:false,
        virtualConsole: jsdom.createVirtualConsole().sendTo(console),
        features : {
            FetchExternalResources: ['script'],
            ProcessExternalResources: ['script'],
            SkipExternalResources: /bootstrap.js/,
            MutationEvents: '2.0'
        },
        resourceLoader: function (resource, callback) {
            var pathname = resource.url.pathname;
            console.log(resource.url.pathname);//
            if (/\/js\/site-/.test(pathname)) {
                console.log(resource.url.pathname);
              resource.url.pathname = pathname.replace("js/", docRoot + "js/");
              return resource.defaultFetch(function (err, body) {
                if (err) return callback(err);
                callback(null, '"use strict";\n' + body);
              });
            } else {
              return callback('');
            }
        },
        created: function(errors, window) {
            window.document.addEventListener('created', function(e) {
                console.log('content loaded');
                pageContent("<!DOCTYPE html>\n"+window.document.documentElement.outerHTML);
               window.close();   
            }, false);
        }
    };
    options.url = serverUrl + action
    var doc = jsdom.jsdom(indexFile, options);
    var window = doc.defaultView;

**Comment by domenic** [comment]
Hmm, this example is not very minimal, as it requires at least one external file and seems to have a lot of complexity.

Can you trim down the parts that are not necessary to reproduce the error? For example, does this still occur if you use something simple like `<!DOCTYPE html><p>Hello</p>` as the string, instad of reading the string from `index.html`? Does this still occur with no resourceLOader? What about with no created handler? What if you remove the `features`? (Including `MutationEvents`, which does nothing). Is turning off strictSSL important?

Basically, please try to get this down to the minimal repro possible. Otherwise this will be hard for us to debug, and thus lower-priority than other things which are easier to debug.
